kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/OperatorMain.java

```diff
@@ -24,30 +30,59 @@
 public class OperatorMain {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(OperatorMain.class);
+    private MeterRegistry registry;
+    private final Operator operator;
+
+    public OperatorMain() {
+        operator = new Operator(o -> o.withMetrics(enablePrometheusMetrics()));
+    }
 
     public static void main(String[] args) {
         try {
-            run();
+            new OperatorMain().run();
         }
         catch (Exception e) {
             LOGGER.error("Operator has thrown exception during startup. Will now exit.", e);
             System.exit(1);
         }
     }
 
-    static void run() {
-        Operator operator = new Operator();
+    void run() {
         operator.installShutdownHook(Duration.ofSeconds(10));
         var registeredController = operator.register(new ProxyReconciler(runtimeDecl()));
         // TODO couple the health of the registeredController to the operator's HTTP healthchecks
         operator.start();
         LOGGER.info("Operator started.");
     }
 
+    public void close() {
```

Comment:
I haven't implemented `closeable` or `autocloseable` for now as I'm not sure the OperatorMain should be either those things. 

Actually given I have added `getRegistry` this could be moved to teardown in the test.